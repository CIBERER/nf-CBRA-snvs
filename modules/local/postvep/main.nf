process POSTVEP {
    tag "${meta.id}"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://docker.io/yolandabq/post_vep:v1' :
        'docker.io/yolandabq/post_vep:v1' }"

    input:
    tuple val(meta), path(vep_tsv), path(roh_automap)
    val maf
    val assembly

    output:
    tuple val(meta), path("*.SNV.INDEL.annotated.tsv"), emit: pvm_tsv

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def automap = roh_automap ? "--automap ${roh_automap}" : ''

    """
    postVEP_modification.R \\
    --input ${vep_tsv} \\
    --output ${prefix}.${assembly}.SNV.INDEL.annotated.tsv \\
    --maf ${maf} \\
    ${automap}

    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bcftools: \$(bcftools --version 2>&1 | head -n1 | sed 's/^.*bcftools //; s/ .*\$//')
    END_VERSIONS
    """
}