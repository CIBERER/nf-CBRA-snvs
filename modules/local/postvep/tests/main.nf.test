nextflow_workflow {

    name "Test Subworkflow POSTVEP"
    script "../main.nf"
    workflow "POSTVEP"
    config "./nextflow.config"

    tag "modules"
    tag "modules/local/postvep"

    test("postvep - vcf ") {

        
        when {
            workflow {
                
                """
                input[0] = Channel.of([
                    [ id:'test'], // meta map
                    file("https://zenodo.org/records/15738776/files/NA12878-HG001.vep.tab.gz?download=1", checkIfExists: true),
                    file("https://zenodo.org/records/15738776/files/NA12878-HG001.HomRegions.tsv?download=1", checkIfExists: true)
                ])
                input[1] = 0.01
                input[2] = "hg38"
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("postvep with no AUTOMAP - vcf ") {

        
        when {
            workflow {
                
                """
                input[0] = Channel.of([
                    [ id:'test'], // meta map
                    file("https://zenodo.org/records/15738776/files/NA12878-HG001.vep.tab.gz?download=1", checkIfExists: true),
                    []
                ])
                input[1] = 0.01
                input[2] = "hg38"
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}