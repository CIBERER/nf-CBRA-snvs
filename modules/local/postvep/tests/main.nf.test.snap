{
    "postvep - vcf ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.hg38.SNV.INDEL.annotated.tsv:md5,c244add25d4b81c33a3cc417f5752e48"
                    ]
                ],
                "pvm_tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.hg38.SNV.INDEL.annotated.tsv:md5,c244add25d4b81c33a3cc417f5752e48"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-06-25T14:06:07.255737431"
    },
    "postvep with no AUTOMAP - vcf ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.hg38.SNV.INDEL.annotated.tsv:md5,83985c01c76cc48c05108f4360e2fe31"
                    ]
                ],
                "pvm_tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.hg38.SNV.INDEL.annotated.tsv:md5,83985c01c76cc48c05108f4360e2fe31"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-06-25T14:06:29.432338863"
    }
}